go/adbc/driver/databricks/error.go

```diff
@@ -0,0 +1,31 @@
+package databricks
+
+import (
+    "regexp"
+    "strings"
+
+    "github.com/apache/arrow-adbc/go/adbc"
+)
+
+const unknownVendorCode = int32(-1) // sentinel value for vendor code; databricks doesn't populate
+
+var (
+    sqlstateRE               = regexp.MustCompile(`(?i)SQLSTATE:\s*([A-Z0-9]{5})`)
```

Comment:
This _feels_ right. Don't make it too free since if it fails, we'll still get a generic error string and the SQLSTATE code in the overall message. Trying to be too strict might bring in malformed codes upon, say, upstream changes.